ext/winevt/winevt_utils.cpp

```diff
@@ -596,8 +596,57 @@ get_description(EVT_HANDLE handle, LANGID langID, EVT_HANDLE hRemote)
   return _wcsdup(result.data());
 }
 
+static int ExpandSIDWString(PSID sid, CHAR **out_expanded)
+{
+#define MAX_NAME 256
+  DWORD len = MAX_NAME, err = ERROR_SUCCESS;
+  SID_NAME_USE sid_type = SidTypeUnknown;
+  char account[MAX_NAME];
+  char domain[MAX_NAME];
+  DWORD result_len = 0;
+  CHAR *formatted = NULL;
+  VALUE vformatted;
+
+  if (!LookupAccountSidA(NULL, sid,
+                         account, &len, domain,
+                         &len, &sid_type)) {
+    err = GetLastError();
+    if (err == ERROR_NONE_MAPPED) {
+      goto not_mapped_error;
+    }
+    else {
+      return -2;
```

Comment:
Does it intentional? (just borrowed from fluent-bit?)

Reply:
Yes. This is intentional. This is because this translation failure shouldn't be handled as an exception.
There is a possibility to happen due to connecting remote session.

Reply:
When returned non-zero from this function, `User` key will not be created: https://github.com/fluent-plugins-nursery/winevt_c/pull/46/files#diff-7d3a7f81db65ff315b6f1e9e118b13ac952f66cc81fae2f05b10c8c6a4c69ac8R897

Reply:
@cosmo0920 
I think he is concerned about the point of returning raw integer, not BOOL, enum, etc.

Reply:
Destroying information into boolean is not good idea. Because the competitor provides both of options `ignore_missing` and `ignore_failure`.
Currently, both of errors will be ignored. In the future enhancements, we'll want to provide options or raising exceptions. So, we need to keep information and shouldn't return boolean here.

Reply:
I see.
I'm not sure (I'm not good at C-programming), but does making `-1` or `-2` an identifier by using `#define` increase code maintainability?

Reply:
Done by https://github.com/fluent-plugins-nursery/winevt_c/pull/46/commits/c907012df8a843569d4dd3a1743fb5e89b65e11b.

Reply:
Thanks!